darts/tests/test_backtesting.py

```diff
@@ -53,6 +60,38 @@ class BacktestingTestCase(unittest.TestCase):
     def setUpClass(cls):
         logging.disable(logging.CRITICAL)
 
+    def test_backtest_forecasting(self):
+        linear_series = lt(length=50)
+
+        # univariate series
+        pred = backtest_forecasting(linear_series, NaiveDrift(), pd.Timestamp('20000201'), 3)
+        self.assertEqual(r2_score(pred, linear_series), 1.0)
+
+        # multivariate series
+        with self.assertRaises(AssertionError):
+            backtest_forecasting(linear_series.stack(linear_series), NaiveDrift(), pd.Timestamp('20000201'), 3)
+
+    def test_backtest_regression(self):
+        gaussian_series = gt(mean=2, length=50)
+        sine_series = st(length=50)
+        features = [gaussian_series + sine_series, gaussian_series]
+        features_multivariate = [(gaussian_series + sine_series).stack(gaussian_series), gaussian_series]
+        target = st(length=50)
+
+        # univariate feature test
+        pred = backtest_regression(features, target, StandardRegressionModel(15), pd.Timestamp('20000201'), 3)
+        self.assertEqual(r2_score(pred, target), 1.0)
```

Comment:
I'm somewhat surprised we get *exactly* a R2 score of 1.0. I get that the model should learn to denoise perfectly, but I would expect the numeric procedure of fitting the model coefficients to converge only with some limited numeric accuracy.

Reply:
So in this case we are using `sklearn.linear_model.LinearRegression`, which is a wrapper around `scipy.linalg.lstsq`, which is simply a solver for the ordinary least squares problem, which I assume is computed in a closed form (that has an ideal solution in this case). I guess this way the numeric accuracy should be pretty good?

Reply:
Yes probably, you're right. Still it's quite neat to see that's exact in this case!